Stripe/STPPaymentIntent.m

```diff
@@ -81,10 +81,6 @@ + (nullable NSString *)idFromClientSecret:(NSString *)clientSecret {
 
 + (STPPaymentIntentStatus)statusFromString:(NSString *)string {
     NSDictionary<NSString *, NSNumber *> *map = @{
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated"
-                                                  @"requires_source": @(STPPaymentIntentStatusRequiresSource),
```

Comment:
Do you have other work I haven't seen that uses the newer API version? I thought it was still possible to receive the `requires_source` status string from the Stripe API, just that we'd want to transparently map that to the `.requiresPaymentMethod` case?

Reply:
You're right, I confused myself here.  We're still on the 2015 API, so the API will send down "requires_source" instead of "requires_payment_method"...